app/src/main/java/com/github/sdp/mediato/utility/adapters/UserAdapter.java

```diff
@@ -1,27 +1,32 @@
 package com.github.sdp.mediato.utility.adapters;
 
+import static com.github.sdp.mediato.data.UserDatabase.followUser;
+import static com.github.sdp.mediato.data.UserDatabase.unfollowUser;
+
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
 import android.os.Handler;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
 import android.widget.ImageView;
 import android.widget.TextView;
+
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
+
 import com.github.sdp.mediato.R;
 import com.github.sdp.mediato.data.UserDatabase;
 import com.github.sdp.mediato.model.User;
 import com.github.sdp.mediato.ui.viewmodel.UserViewModel;
+
 import java.util.concurrent.CompletableFuture;
 
-/**
- * Adapter to display a User list item, displays name and profile pic only.
- */
+
```

Comment:
Why did you revert this back to only have one UserAdapter? I added the UserFollowAdapter  inheriting from UserAdapter so that we could display a slightly different view in the Following and Followers fragment. The UserAdapter superclass takes care of displaying the profile picture and username. Then in the Following fragment, we can extend this to display a button to unfollow the user. And in the Followers fragment, for now there is no button as it uses the UserAdapter superclass (see this [PR](https://github.com/MediaTo-SDP/MediaTo/pull/124)) but in the future we could add another adapter that displays a remove button, like on Instagram.

Reply:
Because for the moment we don't need it. And with your change we weren't able to follow back someone. And if we add later a button to block someone (remove it from our followers and prevent it to follow us) we will also need it everywhere (search, myfollowing, myfollowers).

Reply:
If there is more than two buttons configuration, storing the configuration in an enum and accessing it throigh a switch might be better for code readability.

Reply:
for the moment there is just follow and unfollow buttons. If in a future Sprint, there will be more buttons configurations then yes, a refactoring will be needed.

Reply:
@simone-kalbermatter yep, sorry to have asked for the useless class at the end 😕